Readme.md

```diff
@@ -1,12 +1,528 @@
+
+If you want to see Garmadon in action, have a look at [the main features](#main-features). 
+
+We constantly think of new way to make the best use of all that information that is now available and invite you to do so !
+See [Contributing](#Contributing).
+
+## Table of content
+
+- [Architecture overview](#architecture-overview)
+- [How to install](#how-to-install)
+  * [Requirements](#requirements)
+  * [Backend services sizing](#backend-services-sizing)
+  * [Install Garmadon Reader for Elastic Search](#install-garmadon-reader-for-elastic-search)
+  * [Install Garmadon forwarder](#install-garmadon-forwarder)
+  * [Install Garmadon agent](#install-garmadon-agent)
+  * [Checking your install](#checking-your-install)
+- [Main features](#main-features)
+  * [Dashboarding](#dashboarding)
+    + [End users dashboards](#end-users-dashboards)
+      - [Garmadon-yarn-application](#garmadon-yarn-application)
+      - [Garmadon-standalone-jvm](#garmadon-standalone-jvm)
+      - [Garmadon-spark-job](#garmadon-spark-job)
+      - [Garmadon-spark-job-executors](#garmadon-spark-job-executors)
+      - [Garmadon-spark-job-stages](#garmadon-spark-job-stages)
+    + [SRE dashboard](#sre-dashboard)
+      - [Garmadon-compute](#garmadon-compute)
+      - [Garmadon-hdfs](#garmadon-hdfs)
+      - [Garmadon-server-overview](#garmadon-server-overview)
+  * [Heuristics for DrElephant integration](#heuristics-for-drelephant-integration)
+  * [HDFS dump for batch processing](#hdfs-dump-for-batch-processing)
+  * [Other features](#other-features)
+- [Release management](#release-management)
+- [Monitoring Garmadon](#monitoring-garmadon)
+- [Building Garmadon](#building-garmadon)
+- [Contributing](#contributing)
+
+## Architecture overview
+
+Every hadoop container runs an agent that produces events. Those events are gathered by a process running on each cluster node (the forwarder) and sent to kafka.
+
+They are then consumed by so-called readers to feed other systems (ElasticSearch, DrElephant, HDFS, etc...)
+
+The agent is also deployed on NodeManagers, ResourceManagers and standalone JVMs to provided other types of events (HDFS accesses, container resource tracking, application state tracking, etc...)
+
+![](doc/images/architecture/garmadon_architecture_overview.png)
+
+The agent itself has a very low footprint on running containers. 
+
+![](doc/images/architecture/garmadon_agent_architecture_overview.png)
+
+## How to install
+
+We recommend to use the LATEST version of Garmadon (1.0.0)
+
+### Requirements
+
+Garmadon is designed for **Hadoop 2.6**. You can probably use Garmadon on other Hadoop 2 clusters but we cannot guarantee that for sure.
+
+Garmadon integrates with any JVM based Yarn framework (mapreduce v1, v2, spark, flink, hive, etc... but you won't get information directly from python containers for instance) and has specific extensions for **Spark 2**.
```

Comment:
For non JVM containers it will still provide metrics from nodemanagers/resourcemanager